### PR TITLE
I1070 collect dqm result

### PR DESCRIPTION
**Pull request by Kai-Chen**
Still missing a functional test to check that all dqm validation results are collected in a run.
But that will take some time to build, and I will try to get to it tomorrow -- if not, definitely on Wednesday.

This should be enough to get the next part started, though.

### CODE REVIEW

**Comment by Kai-Chen** [comment]
Just added the missing functional test.
The Scala side for the run collector is now complete.

File: src/main/scala/org/tresamigos/smv/SmvDataSet.scala [hunk 1]
```diff
@@ -447,20 +447,25 @@ abstract class SmvDataSet extends FilenamePart {
...
 
-  private[smv] def computeRDD(genEdd: Boolean): DataFrame = {
+  private[smv] def computeRDD(genEdd: Boolean, collector: SmvRunInfoCollector): DataFrame = {
     val dqmValidator  = new DQMValidator(dqmWithTypeSpecificPolicy(dqm()), isPersistValidateResult)
 
-    if (isEphemeral) {
-      val df = dqmValidator.attachTasks(doRun(dqmValidator))
-      dqmValidator.validate(df, false, moduleValidPath()) // no action before this point
+    // shared logic when running ephemeral and non-ephemeral modules
+    def runDqmAndMeta(df: DataFrame, hasAction: Boolean): Unit = {
```
Comment (AliTajeldin) [inline src/main/scala/org/tresamigos/smv/SmvDataSet.scala]: nice!

**Comment by AliTajeldin** [comment]
oops, CI Validation is failing with the following error:
```
py4j.Py4JException: Method doRun([class org.tresamigos.smv.dqm.DQMValidator]) does not exist
```

**Comment by Kai-Chen** [comment]
Doh, I was hoping to patch the Python API in a separate PR.
Apparently it's gotta go in this one.

**Comment by AliTajeldin** [comment]
We can just do a placeholder for now (e.g. `new RunInfoCollector()` in the proxy).

**Comment by Kai-Chen** [comment]
Yep.
I will do a quick patch.
Coming soon.

**Comment by Kai-Chen** [comment]
Travis doesn't like spark :(

**Comment by AliTajeldin** [comment]
yah, but now it is dying due to resource allocation.  Will merge.

**Pull request merged**